src/ResourceManager/Profile/Commands.Profile/Default/GetAzureRmDefault.cs

```diff
@@ -0,0 +1,67 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Profile.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Management.Internal.Resources;
+using Microsoft.Azure.Management.Internal.Resources.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Profile.Default
+{
+    /// <summary>
+    /// Cmdlet to get default options. 
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmDefault", DefaultParameterSetName = ResourceGroupParameterSet)]
+    [OutputType(typeof(ResourceGroup))]
+    public class GetAzureRMDefaultCommand : AzureRMCmdlet
+    {
+        private const string ResourceGroupParameterSet = "ResourceGroup";
+
+        [Parameter(ParameterSetName = ResourceGroupParameterSet, Mandatory = false, HelpMessage = "Display Default Resource Group", ValueFromPipelineByPropertyName = true)]
+        public SwitchParameter ResourceGroup { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            IAzureContext context = AzureRmProfileProvider.Instance.Profile.DefaultContext;
+            IResourceManagementClient client = AzureSession.Instance.ClientFactory.CreateCustomArmClient<ResourceManagementClient>(
```

Comment:
We should simply use the CreateServiceClient call here - we only need to pvide the context and the name of the resourecemanager endpiut.  This call: https://github.com/Azure/azure-powershell/blob/preview/src/Common/Commands.Common.Authentication.Abstractions/Interfaces/IClientFactory.cs#L39

Reply:
Will stay the same as discussed offline